Geometry/HcalCommonData/src/HcalDDDRecConstants.cc

```diff
@@ -335,10 +335,22 @@ unsigned int HcalDDDRecConstants::nCells() const {
 
 void HcalDDDRecConstants::initialize(void) {
 
+  //Eta grouping
+  int nEta  = (int)(hpar->etagroup.size());
+  if (nEta != (int)(hpar->phigroup.size())) {
+    edm::LogError("HCalGeom") << "HcalDDDRecConstants: sizes of the vectors "
+                              << " etaGroup (" << nEta << ") and phiGroup ("
+                              << hpar->phigroup.size() << ") do not match";
+    throw cms::Exception("DDException") << "HcalDDDRecConstants: inconsistent array sizes" << nEta << ":" << hpar->phigroup.size();
+  }
+
   // First eta table
-  ietaMap.clear(); etaSimValu.clear();
+  iEtaMin     = hpar->etaMin;
+  iEtaMax     = hpar->etaMax;
+  etaTable.clear(); ietaMap.clear(); etaSimValu.clear();
   int ieta(0), ietaHB(0), ietaHE(0);
-  for( unsigned int i=0; i<hpar->etagroup.size(); ++i ) {
+  etaTable.push_back(hpar->etaTable[ieta]);
```

Comment:
@bsunanda - what is the reason for caching the vector locally? nEta is initialised as it size and never changed, so etaTable is always the same as hpar->etaTable